crates/rpc/src/v02/types.rs

```diff
@@ -526,29 +529,43 @@ pub mod reply {
                         }
                     }
                 }
-                starknet_gateway_types::reply::transaction::Transaction::Declare(txn) => {
+                GatewayTransaction::Declare(GatewayDeclare::V0(txn)) => {
                     Self::Declare(DeclareTransaction {
                         common: CommonTransactionProperties {
                             hash: txn.transaction_hash,
                             max_fee: txn.max_fee,
-                            version: txn.version,
+                            version: TransactionVersion::ZERO,
                             signature: txn.signature.clone(),
                             nonce: txn.nonce,
                         },
                         class_hash: txn.class_hash,
                         sender_address: txn.sender_address,
                     })
                 }
-                starknet_gateway_types::reply::transaction::Transaction::Deploy(txn) => {
-                    Self::Deploy(DeployTransaction {
-                        hash: txn.transaction_hash,
+                GatewayTransaction::Declare(GatewayDeclare::V1(txn)) => {
+                    Self::Declare(DeclareTransaction {
+                        common: CommonTransactionProperties {
+                            hash: txn.transaction_hash,
+                            max_fee: txn.max_fee,
+                            version: TransactionVersion::ONE,
+                            signature: txn.signature.clone(),
+                            nonce: txn.nonce,
+                        },
                         class_hash: txn.class_hash,
-                        version: txn.version,
-                        contract_address_salt: txn.contract_address_salt,
-                        constructor_calldata: txn.constructor_calldata.clone(),
+                        sender_address: txn.sender_address,
                     })
                 }
-                starknet_gateway_types::reply::transaction::Transaction::DeployAccount(txn) => {
+                GatewayTransaction::Declare(GatewayDeclare::V2(_)) => {
+                    todo!("v0.11.0: once v1 declare is implemented in RPC")
```

Comment:
They're staying until the RPC work gets done. They should effectively be no-ops since no V2 declare will arrive before v0.11.0 starknet is released. And since we need to fix this before then it doesn't matter which order it is done in.